index.d.ts

```diff
@@ -0,0 +1,28 @@
+import { Linter, Rule } from 'eslint';
+
+declare const plugin: {
+  meta: {
+    name: string;
+    version: string;
+  };
+  environments: {
+    globals: {
+      globals: {
+        [key: string]: boolean;
```

Comment:
note: ideally this should be `Linter.GlobalConf`, but that was introduced in `@types/eslint@9`